### PR TITLE
Extend DataCollatorForCompletionOnlyLM to support correct tool result masking

### PR DESCRIPTION
**Pull request by shanghongsim**
# Description

## Context

Currently, the completions only collator does not consistently mask tool responses (Role.TOOL) correctly. For some chat templates (like Qwen), it accidentally gets it correct, while masking is consistently wrongly applied for llama chat templates. This PR enables the masking to be correctly applied for all chat templates, supporting all four roles (system, user, assistant and tool).

Without `instruction_template`, tool results are masked correctly. But with `instruction_template`, things get weird and some tool results are unmasked. Omitting `instruction_template` causes the collator to find the last `response_template` and masks everything before it (only the final assistant turn trains). With `instruction_template`, the collator searches for `instruction_template` and `response_template` pairs and masks everything in between.

**Case 1: without `instruction_template`**

```
RESPONSE_TEMPLATE = "<|im_start|>assistant\n"

old_no_inst = DataCollatorForCompletionOnlyLM(
    response_template=RESPONSE_TEMPLATE,
    instruction_template=None,
    tokenizer=tokenizer,
)
b = old_no_inst.torch_call([token_ids])
labels_A = b["labels"][0].tolist()
summarise("Case A", labels_A, N)
show_masking(b["input_ids"][0].tolist(), labels_A, tokenizer)
```

image

Without inst template, collator masks everything before the last resp template. So loss only sees

```
<|im_start|>assistant
The weather in Paris is sunny and 18°C.<|im_end|>
```

**Case 2: with `instruction_template`**

With `instruction_template`, it masks everything between a inst and resp template. Since there isn't an inst template before the tool result, it does not get masked properly. 

```
INSTRUCTION_TEMPLATE = "<|im_start|>user\n"

old_with_inst = DataCollatorForCompletionOnlyLM(
    response_template=RESPONSE_TEMPLATE,
    instruction_template=INSTRUCTION_TEMPLATE,
    tokenizer=tokenizer,
)
b2 = old_with_inst.torch_call([token_ids])
labels_B = b2["labels"][0].tolist()
summarise("Case B", labels_B, N)
show_masking(b2["input_ids"][0].tolist(), labels_B, tokenizer)
```

image

In case 2, the loss sees tool result, which is incorrect.

```
...
<|im_start|>assistant
<tool_call>
{"name": "get_weather", "arguments": {"location": "Paris"}}
</tool_call><|im_end|>
...
<|im_start|>assistant
The weather in Paris is sunny and 18°C.<|im_end|>
```

To confirm my hypothesis of no inst template before the tool result being the cause, I experimented with adding a user turn before the tool call and masking works correctly in that case.

image

## ToolAwareCompletionsCollator
With the new `ToolAwareCompletionsCollator`, tool results are masked properly

Without `instruction_template`
image

With `instruction_template`
image

## Change 1: Extend `DataCollatorForCompletionOnlyLM` with span-based masking
Extend `DataCollatorForCompletionOnlyLM` with span-based masking (detecting assistant turn boundaries via response + end-of-turn tokens) beyond instruction based masking (matching instruction/response string pairs) supported currently. 

## Change 2: `train_target` enum for explicit control of which part of response to train on
- **`assistant_turn`**: Masks everything, then unmasks each assistant response bounded by `response_template` .. `end_of_turn_template` (inclusive of EOT). 
- **`final_assistant_turn`**: Masks all tokens before the last `response_template` occurrence. 

When `train_target` is set, no matter what templates have been provided, `train_target` has final authority on which combination is used. When `train_target` is not set, it is inferred from the templates provided. Eventually, we want users to specify `train_target` instead of relying on combinations of templates to control behavior. However, since we have many configs in production that do not use `train_target` and set templates explicitly, this shall be the interim solution to ensure old configs do not break. Existing configs using `collator_kwargs` with `instruction_template` + `response_template` continue to work via the legacy path (with a deprecation warning). New configs should use `train_target` explicitly. Users can set `train_target` and override the templates manually by setting `collator_kwargs`.

**Before** — users must provide model-specific token strings:
```yaml
collator_name: "text_completions_only_with_padding"
collator_kwargs:
  response_template: "<|im_start|>assistant\n"
  end_of_turn_template: "<|im_end|>"
```

**After** — users express intent, templates auto-resolve from tokenizer:
```yaml
collator_name: "text_completions_only_with_padding"
train_target: "assistant_turn"
```

## Change 3: Auto detection of collator templates
Instead of fragile vocab and collator template matching (requires us to maintain collator templates for all popular models), shift to a more robust approach: apply the chat template to a known test conversation, then finds the assistant boundary strings in the rendered output.

## Migration and deprecation considerations:
1. Configs that **only** specify `collator_name` -> removed support for this with the  removal of the default llama template fallback. This should be ok as no configs in production specify `collator_name` without `collator_kwargs`. Only less than 10 configs in OSS fall in this edgecase and they are being updated in this PR.
2. Enterprise SFT configs -> they should still work with old legacy behavior. Will update them once OSS version is updated in API.

## Open issues for next time
- Move`train_target` into `collator_kwargs` -> I considered putting train_target inside collator_kwargs instead of making it a separate field, but collator_kwargs is an opaque dict. Users wouldn't know train_target exists unless they read the docs. A top-level field makes it more visible. This will eventually be solved when we shift to `CollatorParam`.
- `CollatorParams`: This would mean changing every YAML from:

```
  collator_name: "text_completions_only_with_padding"
  collator_kwargs:
    train_target: "all_assistant_turns"
```
to 

```
  collator:
    name: "text_completions_only_with_padding"
    train_target: "all_assistant_turns"
```

This is much cleaner but its a substantial refactor. In order to manage the scope of this PR, this shall be deferred to a future PR
- Vision specific concerns -> future PR to neaten the vision concerns that are now scattered throughout the LM specific things
- Per-split collator configuration -> `build_collator_from_config` builds a single collator from the training split's settings, which is reused for all splits (train, validation, test). Validation/test split-specific collator settings (e.g. a different `train_target`) are silently ignored. This is pre-existing behavior not introduced by this PR, but worth noting as a known limitation. Fixing it would require refactoring the training loop to build per-split collators.

# Related issues

Fixes https://linear.app/oumi/issue/OPE-1870

# Before submitting

- [ ] This PR only changes documentation.
- [x] Did you read the [contributor guideline](https://github.com/oumi-ai/oumi/blob/main/CONTRIBUTING.md) Pull Request guidelines?
- [ ] Did you link the issue(s) related to this PR in the section above?
- [x] Did you add / update tests where needed?

# Reviewers

At least one review from a member of `oumi-ai/oumi-staff` is required.

### CODE REVIEW

File: src/oumi/builders/collators.py [hunk 1]
```diff
@@ -127,33 +208,15 @@ def build_data_collator(
...
+            raise ValueError(
+                "response_template is required for "
+                "'text_completions_only_with_padding'. Provide it via "
+                "collator_kwargs or use masking_method for auto-resolution."
             )
-
-        if not response_template:
-            response_template = "<|start_header_id|>assistant<|end_header_id|>\n\n"
-
         return TextCompletionsCollatorWithPadding(
```
[the remark this answers is not in the document]
Comment (shanghongsim) [reply src/oumi/builders/collators.py:241]: fixed with new commit

File: src/oumi/builders/collators.py [hunk 2]
```diff
@@ -127,33 +208,21 @@ def build_data_collator(
...
             tokenizer=tokenizer,
-            response_template=response_template,
-            instruction_template=instruction_template,
             debug=debug,
```
[the remark this answers is not in the document]
Comment (shanghongsim) [reply src/oumi/builders/collators.py]: Fixed. Now, if user only specifies `collator_name` without `collator_kwargs` or `masking_method`, ity will fallback to legacy behavior with llama defaults.

File: src/oumi/builders/collators.py [hunk 3]
```diff
@@ -127,33 +208,39 @@ def build_data_collator(
...
+        # span-based masking (end_of_turn_template makes instruction_template
+        # unnecessary since masking is handled by response/eot spans).
+        if kwargs.get("end_of_turn_template") is None:
+            if not kwargs.get("instruction_template"):
+                logger.warning(
+                    "No instruction_template provided for "
+                    "'text_completions_only_with_padding'. "
+                    "Falling back to Llama-style default. "
+                    "Please set instruction_template via collator_kwargs "
+                    "or use masking_method for auto-resolution."
```
[the remark this answers is not in the document]
Comment (shanghongsim) [reply src/oumi/builders/collators.py]: Fixed. Defaults only applied when `masking_method` is `None`

File: src/oumi/builders/collators.py [hunk 4]
```diff
@@ -127,33 +208,39 @@ def build_data_collator(
...
-        if end_of_turn_template is None:
-            instruction_template = (
-                instruction_template
-                if instruction_template
-                else "<|start_header_id|>user<|end_header_id|>\n\n"
-            )
+        # span-based masking (end_of_turn_template makes instruction_template
+        # unnecessary since masking is handled by response/eot spans).
+        if kwargs.get("end_of_turn_template") is None:
+            if not kwargs.get("instruction_template"):
```
[the remark this answers is not in the document]
Comment (shanghongsim) [reply src/oumi/builders/collators.py]: Fixed. Defaults only applied when `masking_method` is `None`

File: src/oumi/builders/collators.py [hunk 5]
```diff
@@ -127,33 +208,42 @@ def build_data_collator(
...
+                        "<|start_header_id|>user<|end_header_id|>\n\n",
+                    )
 
-        if not response_template:
-            response_template = "<|start_header_id|>assistant<|end_header_id|>\n\n"
+            if not kwargs.get("response_template"):
+                logger.warning(
+                    "No response_template provided for "
+                    "'text_completions_only_with_padding'. "
+                    "Falling back to Llama-style default. "
```
[the remark this answers is not in the document]
Comment (shanghongsim) [reply src/oumi/builders/collators.py]: 1. There's no path where masking_method arrives in kwargs without response_template. _build_masking_kwargs always includes it 
2. masking_method is not situated in collator_kwargs, they are on the same level

[the remark this answers is not in the document]
Comment (shanghongsim) [reply src/oumi/builders/collators.py]: `build_data_collator` is only access by `builc_collator_from_config` and it never calls `build_data_collator` with `masking_method` and without `response_template`. If there is `masking_method`, there will be `response_template`. This is supplied by `_build_masking_kwargs`.

File: src/oumi/builders/collators.py [hunk 6]
```diff
@@ -27,11 +28,101 @@
...
+    between_ids = tokenizer.encode(
+        rendered[first_asst_end:second_user], add_special_tokens=False
+    )
+    eot_len = 0
+    for a, b in zip(after_ids, between_ids):
+        if a != b:
+            break
+        eot_len += 1
+    eot_ids = after_ids[:eot_len]
+    end_of_turn_template: str = tokenizer.decode(eot_ids, skip_special_tokens=False)
```
[the remark this answers is not in the document]
Comment (shanghongsim) [reply src/oumi/builders/collators.py]: fixed

File: src/oumi/builders/collators.py [hunk 7]
```diff
@@ -27,11 +27,101 @@
...
+    between_ids = tokenizer.encode(
+        rendered[first_asst_end:second_user], add_special_tokens=False
+    )
+    eot_len = 0
+    for a, b in zip(after_ids, between_ids):
+        if a != b:
+            break
+        eot_len += 1
+    eot_ids = after_ids[:eot_len]
+    end_of_turn_template = tokenizer.decode(eot_ids, skip_special_tokens=False)
```
[the remark this answers is not in the document]
Comment (shanghongsim) [reply src/oumi/builders/collators.py]: fixed

[the remark this answers is not in the document]
Comment (shanghongsim) [reply src/oumi/builders/collators.py]: fixed

File: src/oumi/builders/collators.py [hunk 8]
```diff
@@ -27,11 +27,101 @@
...
+    between_ids = tokenizer.encode(
+        rendered[first_asst_end:second_user], add_special_tokens=False
+    )
+    eot_len = 0
+    for a, b in zip(after_ids, between_ids):
+        if a != b:
+            break
+        eot_len += 1
+    eot_ids = after_ids[:eot_len]
+    end_of_turn = tokenizer.decode(eot_ids, skip_special_tokens=False)
```
[the remark this answers is not in the document]
Comment (shanghongsim) [reply src/oumi/builders/collators.py]: fixed

File: src/oumi/builders/collators.py [hunk 9]
```diff
@@ -27,11 +27,100 @@
...
+
```
[the remark this answers is not in the document]
Comment (shanghongsim) [reply src/oumi/builders/collators.py:110]: fixed

File: src/oumi/builders/collators.py [hunk 10]
```diff
@@ -198,6 +290,14 @@ def build_collator_from_config(
...
+    if train_split.train_target is not None:
+        response_template, end_of_turn_template = _resolve_collator_templates(tokenizer)
+        collator_kwargs["response_template"] = response_template
+        collator_kwargs["train_target"] = train_split.train_target.value
+        if train_split.train_target == TrainTarget.ALL_ASSISTANT_TURNS:
+            collator_kwargs["end_of_turn_template"] = end_of_turn_template
+
     # Merge collator_kwargs from config with the existing kwargs
     # Config kwargs take precedence over automatically determined kwargs
     config_collator_kwargs = train_split.collator_kwargs or {}
```
[the remark this answers is not in the document]
Comment (shanghongsim) [reply src/oumi/builders/collators.py]: Preexisting bug, will fix in next PR

File: src/oumi/builders/collators.py [hunk 11]
```diff
@@ -206,9 +291,55 @@ def build_collator_from_config(
...
+            try:
+                response_template, end_of_turn_template = _resolve_collator_templates(
+                    tokenizer
+                )
+                collator_kwargs["response_template"] = response_template
+                if train_split.train_target == TrainTarget.ALL_ASSISTANT_TURNS:
+                    collator_kwargs["end_of_turn_template"] = end_of_turn_template
+            except ValueError:
+                if config_collator_kwargs.get("response_template") is None:
+                    raise
```
[the remark this answers is not in the document]
Comment (shanghongsim) [reply src/oumi/builders/collators.py:325]: Fixed

File: src/oumi/builders/collators.py [hunk 12]
```diff
@@ -27,11 +28,105 @@
...
+    Applies the chat template to a known test conversation, then finds
+    the assistant boundary strings in the rendered output.
+
+    Returns:
+        (response_template, end_of_turn_template)
+
+    Raises:
+        ValueError: If templates cannot be extracted.
+    """
+    msgs = [
```
Comment (jgreer013) [inline src/oumi/builders/collators.py:58]: This is really clever, but we should maybe include a system instruction here?

File: src/oumi/builders/collators.py [hunk 13]
```diff
@@ -27,11 +28,105 @@
...
+    except Exception as exc:
+        raise ValueError(_FALLBACK_MSG) from exc
+
+    if not isinstance(rendered, str):
+        raise ValueError(_FALLBACK_MSG)
+
+    # Locate boundaries around the second turn pair
+    # to avoid system-prompt effects on the first turn.
+    try:
+        a1 = rendered.index(_SENTINEL_ASST)
```
Comment (jgreer013) [inline src/oumi/builders/collators.py:79]: nit: first_asst_start, and add _start to second turns

File: src/oumi/builders/collators.py [hunk 14]
```diff
@@ -27,11 +28,105 @@
...
+        first_asst_end = a1 + len(_SENTINEL_ASST)
+        second_user = rendered.index(_SENTINEL_USER, first_asst_end)
+        second_user_end = second_user + len(_SENTINEL_USER)
+        second_asst = rendered.index(_SENTINEL_ASST, second_user_end)
+        second_asst_end = second_asst + len(_SENTINEL_ASST)
+    except ValueError:
+        raise ValueError(_FALLBACK_MSG)
+
+    # End-of-turn: common token-ID prefix of the two strings that
```
Comment (jgreer013) [inline src/oumi/builders/collators.py:88]: extract this to its own method

File: src/oumi/builders/collators.py [hunk 15]
```diff
@@ -27,11 +28,105 @@
...
+    for a, b in zip(after_ids, between_ids):
+        if a != b:
+            break
+        eot_len += 1
+    eot_ids = after_ids[:eot_len]
+    _eot_decoded = tokenizer.decode(eot_ids, skip_special_tokens=False)
+    assert isinstance(_eot_decoded, str)
+    end_of_turn_template = _eot_decoded
+
+    # Response template: strip the EOT prefix to get just the assistant header.
```
Comment (jgreer013) [inline src/oumi/builders/collators.py:104]: Same here, extract to its own method

File: src/oumi/builders/collators.py [hunk 16]
```diff
@@ -27,11 +28,105 @@
...
+    if eot_len > 0 and resp_ids[:eot_len] == eot_ids:
+        resp_ids = resp_ids[eot_len:]
+    _resp_decoded = tokenizer.decode(resp_ids, skip_special_tokens=False)
+    assert isinstance(_resp_decoded, str)
+    response_template = _resp_decoded
+
+    if not response_template.strip() or not end_of_turn_template.strip():
+        raise ValueError(_FALLBACK_MSG)
+
+    # Qwen3 and similar reasoning models inject <think>...</think> into
```
Comment (jgreer013) [inline src/oumi/builders/collators.py:117]: This description scares me, I wonder if we could have a better workaround or a louder error

File: src/oumi/core/collators/trl_data_collator_for_completion_only_lm.py [hunk 17]
```diff
@@ -19,18 +19,64 @@
 import torch
 from transformers.data.data_collator import DataCollatorForLanguageModeling
 
+from oumi.core.configs.params.data_params import TrainTarget
+
 
 class DataCollatorForCompletionOnlyLM(DataCollatorForLanguageModeling):
```
Comment (jgreer013) [inline src/oumi/core/collators/trl_data_collator_for_completion_only_lm.py:25]: It's not clear to me what this collator does differently than the other one.

**Review by jgreer013 — APPROVED** [review]

File: src/oumi/builders/collators.py (hunk 12 above)
Comment (shanghongsim) [reply src/oumi/builders/collators.py:58]: Like modify it to extract system instruction? We are current no longer relying on system instruction for masking

File: src/oumi/builders/collators.py [hunk 18]
```diff
@@ -27,11 +28,114 @@
...
 _VERY_LARGE_INTEGER = int(1e30)
+_SENTINEL_USER = "<<__U__>>"
+_SENTINEL_ASST = "<<__A__>>"
+_FIX_HINT = (
+    "Fix: provide response_template (and end_of_turn_template for "
+    "all_assistant_turns) in collator_kwargs."
+)
+
+
+def _resolve_collator_templates(
```
Comment (lefft) [inline src/oumi/builders/collators.py:44]: Has this been tested with gpt-oss models? They use a non-standard system for marking boundaries within responses, so wondering if boundary detection will work as it does for Qwen/Llama style templates.

File: src/oumi/builders/collators.py [hunk 19]
```diff
@@ -206,9 +303,66 @@ def build_collator_from_config(
...
+            # Path 2: train_target not set, templates provided manually
+            # via collator_kwargs. Infer train_target from which templates
+            # are present.
+            has_eot = config_collator_kwargs.get("end_of_turn_template") is not None
+            has_inst = config_collator_kwargs.get("instruction_template") is not None
+            if has_eot:
+                collator_kwargs["train_target"] = "all_assistant_turns"
+            elif has_inst:
+                warnings.warn(
+                    "Instruction-based masking is deprecated.\n"
```
Comment (lefft) [inline src/oumi/builders/collators.py:348]: Is there a plan/intention to eliminate support for instruction-based masking altogether, or will we just keep it in a "deprecated" state? 

Want to make sure we retain compatibility with existing enterprise configs (or verify that moving to this style doesn't change training dynamics before adapting enterprise configs).

**Review by lefft — APPROVED** [review]
Can we be sure to check whether template detection and collation happens properly when user or assistant turns begin with (one or more) leading `\n`s? We had issues with this breaking tokenization and leading to records being skipped in enterprise, would like to understand whether these changes will help, harm, or not impact that issue.

File: src/oumi/builders/collators.py (hunk 12 above)
Comment (jgreer013) [reply src/oumi/builders/collators.py:58]: No I mean ensure that when we extract the different template portions, we do so using an example conversation that also includes a system instruction

Or rather, perhaps do it twice, once on a conversation with a system instruction, and once on a conversation without.

Why? Because I'm a bit concerned of the scenario where this logic doesn't use system instructions when extracting, but the user does in their data, and *somehow* the resulting templates we extract using this methodology don't wind up working when the data includes the presence of a system instruction for some reason.

File: src/oumi/core/collators/trl_data_collator_for_completion_only_lm.py (hunk 17 above)
Comment (shanghongsim) [reply src/oumi/core/collators/trl_data_collator_for_completion_only_lm.py:25]: Previously, it would `instruction_template` and `response_template` pairs then mask everything in between (see L324). However, there isn't an `instruction_template` before the tool result, so tool result content does not get masked properly. Some models (like Qwen) *accidentally* mask it correctly because their chat template stores tool result under role.user. But for most models (like Llama etc), tool result is stored under a role.tool with a separate tag so masking is incorrectly applied for the reasons above. The new approach basically masks everything then only unmask portions between the `response_template` and `end_of_turn_template`. This is more robust to the different roles and specifics of how tool results are formatted and stored by different chat templates.

image

**Review by oelachqar — APPROVED** [review]

File: tests/integration/builders/test_collator_template_detection.py [hunk 20]
```diff
@@ -0,0 +1,240 @@
...
+) -> transformers.PreTrainedTokenizerBase:
+    tok = transformers.AutoTokenizer.from_pretrained(
+        model_name, trust_remote_code=trust_remote_code
+    )
+    if not tok.chat_template:
+        try:
+            jinja_path = hf_hub_download(model_name, "chat_template.jinja")
+            with open(jinja_path) as f:
+                tok.chat_template = f.read()
+        except Exception:
```
[the remark this answers is not in the document]
Comment (shanghongsim) [reply tests/integration/builders/test_collator_template_detection.py]: This has been fixed

File: src/oumi/builders/collators.py (hunk 16 above)
Comment (shanghongsim) [reply src/oumi/builders/collators.py]: Added code to strip the <think> in `_detect_response_template`

File: src/oumi/builders/collators.py (hunk 15 above)
Comment (shanghongsim) [reply src/oumi/builders/collators.py]: Done!

File: src/oumi/builders/collators.py (hunk 14 above)
Comment (shanghongsim) [reply src/oumi/builders/collators.py]: Done!

File: src/oumi/builders/collators.py (hunk 13 above)
Comment (shanghongsim) [reply src/oumi/builders/collators.py]: Done!

File: src/oumi/builders/collators.py (hunk 18 above)
Comment (shanghongsim) [reply src/oumi/builders/collators.py]: Yes, tested. There is an integration test in tests/integration/builders/test_collator_template_detection.py

File: src/oumi/builders/collators.py (hunk 19 above)
Comment (shanghongsim) [reply src/oumi/builders/collators.py:415]: Plan is to eliminate instruction-based masking altogether. In the interim, while we do more testing, it will be in deprecated state.

File: src/oumi/builders/collators.py (hunk 12 above)
Comment (shanghongsim) [reply src/oumi/builders/collators.py]: Ok, added!

**Comment by shanghongsim** [comment]
> Can we be sure to check whether template detection and collation happens properly when user or assistant turns begin with (one or more) leading `\n`s? We had issues with this breaking tokenization and leading to records being skipped in enterprise, would like to understand whether these changes will help, harm, or not impact that issue.

- Added `.rstrip("\n")` in `_detect_response_template` (collators.py:119)
- Added two tests to verify correct behavior with leading `\n`: `test_leading_newline_bpe_merge_regression` and `test_span_masking_with_leading_newline_content`

cc @lefft

File: src/oumi/builders/collators.py [hunk 21]
```diff
@@ -27,11 +28,105 @@
...
+        rendered[second_user_end:second_asst], add_special_tokens=False
+    )
+    if eot_len > 0 and resp_ids[:eot_len] == eot_ids:
+        resp_ids = resp_ids[eot_len:]
+    _resp_decoded = tokenizer.decode(resp_ids, skip_special_tokens=False)
+    assert isinstance(_resp_decoded, str)
+    response_template = _resp_decoded
+
+    if not response_template.strip() or not end_of_turn_template.strip():
+        raise ValueError(_FALLBACK_MSG)
```
[the remark this answers is not in the document]
Comment (shanghongsim) [reply src/oumi/builders/collators.py]: We should be able to extract `end_of_turn_template` even when `train_target` is `FINAL_ASSISTANT_TURN`

File: src/oumi/builders/collators.py [hunk 22]
```diff
@@ -126,27 +280,19 @@ def build_data_collator(
...
         return TextCompletionsCollatorWithPadding(
             tokenizer=tokenizer,
-            instruction_prefix=instruction_prefix,
-            response_prefix=response_prefix,
             debug=debug,
+            ignore_index=(
+                label_ignore_index if label_ignore_index is not None else -100
+            ),
             **kwargs,
         )
```
[the remark this answers is not in the document]
Comment (shanghongsim) [reply src/oumi/builders/collators.py:315]: Fixed

File: src/oumi/builders/collators.py [hunk 23]
```diff
@@ -126,27 +231,19 @@ def build_data_collator(
...
     raise ValueError(f"Unknown data collator name: '{collator_name}'")
```
[the remark this answers is not in the document]
Comment (shanghongsim) [reply src/oumi/builders/collators.py:316]: Fixed
